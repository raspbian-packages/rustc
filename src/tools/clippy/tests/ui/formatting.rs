#![feature(plugin)]
#![plugin(clippy)]

#![warn(clippy)]
#![allow(unused_variables)]
#![allow(unused_assignments)]
#![allow(if_same_then_else)]
#![allow(deref_addrof)]

fn foo() -> bool { true }

fn main() {
    // weird `else if` formatting:
    if foo() {
    } if foo() {
    }

    let _ = { // if as the last expression
        let _ = 0;

        if foo() {
        } if foo() {
        }
        else {
        }
    };

    let _ = { // if in the middle of a block
        if foo() {
        } if foo() {
        }
        else {
        }

        let _ = 0;
    };

    if foo() {
    } else
    if foo() { // the span of the above error should continue here
    }

    if foo() {
    }
    else
    if foo() { // the span of the above error should continue here
    }

    // those are ok:
    if foo() {
    }
    if foo() {
    }

    if foo() {
    } else if foo() {
    }

    if foo() {
    }
    else if foo() {
    }

    if foo() {
    }
    else if
    foo() {}

    // weird op_eq formatting:
    let mut a = 42;
    a =- 35;
    a =* &191;

    let mut b = true;
    b =! false;

    // those are ok:
    a = -35;
    a = *&191;
    b = !false;

    // possible missing comma in an array
    let _ = &[
        -1, -2, -3 // <= no comma here
        -4, -5, -6
    ];
    let _ = &[
        -1, -2, -3 // <= no comma here
        *4, -5, -6
    ];

    // those are ok:
    let _ = &[
        -1, -2, -3,
        -4, -5, -6
    ];
    let _ = &[
        -1, -2, -3,
        -4, -5, -6,
    ];
    let _ = &[
        1 + 2, 3 +
        4, 5 + 6,
    ];
}
