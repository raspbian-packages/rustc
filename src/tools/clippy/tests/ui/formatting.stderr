error: this looks like an `else if` but the `else` is missing
  --> $DIR/formatting.rs:15:6
   |
15 |     } if foo() {
   |      ^
   |
   = note: `-D suspicious-else-formatting` implied by `-D warnings`
   = note: to remove this lint, add the missing `else` or add a new line before the second `if`

error: this looks like an `else if` but the `else` is missing
  --> $DIR/formatting.rs:22:10
   |
22 |         } if foo() {
   |          ^
   |
   = note: to remove this lint, add the missing `else` or add a new line before the second `if`

error: this looks like an `else if` but the `else` is missing
  --> $DIR/formatting.rs:30:10
   |
30 |         } if foo() {
   |          ^
   |
   = note: to remove this lint, add the missing `else` or add a new line before the second `if`

error: this is an `else if` but the formatting might hide it
  --> $DIR/formatting.rs:39:6
   |
39 |       } else
   |  ______^
40 | |     if foo() { // the span of the above error should continue here
   | |____^
   |
   = note: to remove this lint, remove the `else` or remove the new line between `else` and `if`

error: this is an `else if` but the formatting might hide it
  --> $DIR/formatting.rs:44:6
   |
44 |       }
   |  ______^
45 | |     else
46 | |     if foo() { // the span of the above error should continue here
   | |____^
   |
   = note: to remove this lint, remove the `else` or remove the new line between `else` and `if`

error: this looks like you are trying to use `.. -= ..`, but you really are doing `.. = (- ..)`
  --> $DIR/formatting.rs:71:6
   |
71 |     a =- 35;
   |      ^^^^
   |
   = note: `-D suspicious-assignment-formatting` implied by `-D warnings`
   = note: to remove this lint, use either `-=` or `= -`

error: this looks like you are trying to use `.. *= ..`, but you really are doing `.. = (* ..)`
  --> $DIR/formatting.rs:72:6
   |
72 |     a =* &191;
   |      ^^^^
   |
   = note: to remove this lint, use either `*=` or `= *`

error: this looks like you are trying to use `.. != ..`, but you really are doing `.. = (! ..)`
  --> $DIR/formatting.rs:75:6
   |
75 |     b =! false;
   |      ^^^^
   |
   = note: to remove this lint, use either `!=` or `= !`

error: possibly missing a comma here
  --> $DIR/formatting.rs:84:19
   |
84 |         -1, -2, -3 // <= no comma here
   |                   ^
   |
   = note: `-D possible-missing-comma` implied by `-D warnings`
   = note: to remove this lint, add a comma or write the expr in a single line

error: possibly missing a comma here
  --> $DIR/formatting.rs:88:19
   |
88 |         -1, -2, -3 // <= no comma here
   |                   ^
   |
   = note: to remove this lint, add a comma or write the expr in a single line

error: aborting due to 10 previous errors

